utils/src/main/scala/za/co/absa/hermes/utils/HelperFunctions.scala

```diff
@@ -38,9 +38,11 @@ object HelperFunctions {
     * especially on a vary big dataframes.
     *
     * @param df A dataframe
+   *  @param dfsForArraysLength DataFrames with the same schema as `df` that are used to
+   *                             flatten arrays with maximum length among `df` and `dfsForArraysLengths`.
     * @return A new dataframe with flat schema.
     */
-  def flattenSchema(df: DataFrame): List[Column] = {
+  def flattenSchema(df: DataFrame, dfsForArraysLength: DataFrame*): List[Column] = {
```

Comment:
Wouldn't be enough to name them `otherDfs` instead of `dfsForArraysLength`?

Reply:
I was thinking about that name, or just `dfs`, but these DataFrames are really just to compute arrays length, and names like `dfs` or `otherDfs` kinda suggest to me that schema for these is also flattened (which is not necessary true).

Reply:
Fair enough